examples/spotify-api-example/src/main/java/com/spotify/apollo/example/AlbumResource.java

```diff
@@ -0,0 +1,109 @@
+package com.spotify.apollo.example;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.spotify.apollo.Client;
+import com.spotify.apollo.Request;
+import com.spotify.apollo.RequestContext;
+import com.spotify.apollo.Response;
+import com.spotify.apollo.Status;
+import com.spotify.apollo.example.data.Album;
+import com.spotify.apollo.example.data.Artist;
+import com.spotify.apollo.route.AsyncHandler;
+import com.spotify.apollo.route.Middlewares;
+import com.spotify.apollo.route.Route;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.StringJoiner;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+
+import okio.ByteString;
+
+/**
+ * The album resource demonstrates how to use asynchronous routes in Apollo.
+ */
+public class AlbumResource {
+
+  private static final String SPOTIFY_API = "https://api.spotify.com";
+  private static final String SEARCH_API = SPOTIFY_API + "/v1/search";
+  private static final String ALBUM_API = SPOTIFY_API + "/v1/albums";
+
+  public Stream<Route<AsyncHandler<Response<ByteString>>>> routes() {
+    // The album resource has two routes. Both are using the autoserializer.
+    return Stream.of(
+        Route.async("GET", "/albums/new", this::getAlbums),
+        Route.async("GET", "/albums/hipster", this::getAlbums)
+    )
+        .map(route -> route.withMiddleware(Middlewares::autoSerialize));
+  }
+
+  public CompletionStage<Response<ArrayList<Album>>> getAlbums(RequestContext requestContext) {
+    String uri = requestContext.request().uri();
+    String tag = uri.substring(uri.lastIndexOf("/") + 1);
```

Comment:
I considered this, but there was two reasons I didn't: 1) To show a resource with multiple routes 2) Because new and hipster are actually the only valid tag values in the Spotify API...

Maybe I could clarify this with a comment, saying that you could do it the other way instead? (parameters are examplified in the other resource)